### ISSUE TITLE
stream response memory leak

### ISSUE THREAD

**Issue by odysseusmax** [aio-libs/aiohttp#3673]
## Long story short
I'm using aiohttp for downloading files. From my experience I found that the chunked stream requests are causing memory leaks. I'm working with large files and i have very limited memory. When i make a request, it completes the request succesfully, but the response is loaded into memory and not released causing memory overflow.

## Expected behaviour

Load chunked responses and release it to avoid memory leak.

## Actual behaviour

Loads response content as chunks and not releasing.

## Steps to reproduce

This is what i use.
```
url = 'some direct download link'
filename = 'some file name'
async with aiohttp.ClientSession() as session:
                async with session.get(url) as r:
                    if(r.status < 400):
                          with open(filename, 'wb') as fd:
                                        async for chunk in r.content.iter_any():
                                            fd.write(chunk)
```

## Your environment
using aiohttp-client
script running on heroku
latest aiohttp (as of now)

**Comment by socketpair** [comment]
Seems, this is https://github.com/aio-libs/aiohttp/issues/3631. Please recheck. If so, please close issue. If not, please provide us a way to reproduce the bug.

**Comment by socketpair** [comment]
@asvetlov interesting, this bug was added a long time before I have started to inspect streams code :)

**Comment by odysseusmax** [comment]
> Seems, this is #3631. Please recheck. If so, please close issue. If not, please provide us a way to reproduce the bug.

The code i use is provided here. The issue you mentioned is with post method, for me its with get method.

**Comment by socketpair** [comment]
@odysseusmax you did not provide exact URL. The issue I have provided is not connected with specific method. It is connected with reading a chunked response. So, GET is affected too.

**Comment by odysseusmax** [comment]
@socketpair any link that points to some file greater than 1GB

**Comment by socketpair** [comment]
@odysseusmax if so, I consider this bug is fixed. Please re-test on latest 4.0 or 3.5 branch.

**Comment by odysseusmax** [comment]
> @odysseusmax if so, I consider this bug is fixed. Please re-test on latest 4.0 or 3.5 branch.

okey will test and respond

**Comment by socketpair** [comment]
Reopen please if required.

**Comment by odysseusmax** [comment]
Issue is solved in 4.0

**Comment by socketpair** [comment]
@odysseusmax actually in latest 3.5 too.
